### PR TITLE
[Components] americommerce

### LINKED ISSUE #14198: [Components] americommerce
# americommerce
## URLs
- https://github.com/AmeriCommerce/rest-api/tree/master
#### Prompt
Emit new event when a customer is created in your AmeriCommerce store. Props: store ID (required), customer ID (optional)
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/tree/master
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/blob/master/.spectral.mjs
- https://github.com/AmeriCommerce/rest-api/actions
- https://github.com/AmeriCommerce/rest-api/activity
#### Prompt
Emit new event when a product is added to your AmeriCommerce store. Props: store ID (required), product ID (optional)
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/tree/master
- https://github.com/AmeriCommerce/rest-api/branches
- https://github.com/AmeriCommerce/rest-api/tags
- https://github.com/AmeriCommerce/rest-api/pulse
- https://github.com/AmeriCommerce/rest-api/commits/master/
- https://github.com/AmeriCommerce/rest-api/tree/master/.vscode
- https://github.com/AmeriCommerce/rest-api/tree/master/docs
- https://github.com/AmeriCommerce/rest-api/tree/master/examples
- https://github.com/AmeriCommerce/rest-api/blob/master/.editorconfig
- https://github.com/AmeriCommerce/rest-api/blob/master/.gitattributes
- https://github.com/AmeriCommerce/rest-api/blob/master/.gitignore
- https://github.com/AmeriCommerce/rest-api/blob/master/.spectral.mjs
- https://github.com/AmeriCommerce/rest-api/blob/master/.stoplight.json
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/blob/master/toc.json
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/datetimes.md
- https://github.com/AmeriCommerce/rest-api/custom-properties
#### Prompt
Emit new event when a new valid order (not declined or cancelled) is created in your AmeriCommerce store. Props: store ID (required), order ID (optional)
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/tree/master
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/blob/master/.stoplight.json
#### Prompt
Changes the status of an existing order. Required props: order_id (the unique identifier for the order), new_status (the new status for the order).
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/tree/master/docs
- https://github.com/AmeriCommerce/rest-api/tree/master/examples
#### Prompt
Creates a new order or updates an existing one. Required props: order_details (the information for the order you're creating or updating). Optional props: customer_id(customer related with the order), payment_method (information about the order's payment method).
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/tree/master
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/blob/master/toc.json
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/datetimes.md
- https://github.com/AmeriCommerce/rest-api/releases
#### Prompt
Updates the details of a registered customer. Required props: customer_id (the unique identifier for the customer), customer_details (the new information for the customer).
#### URLs
- https://github.com/AmeriCommerce/rest-api
- https://github.com/AmeriCommerce/rest-api/blob/master/docs/docs/authentication.md
- https://github.com/AmeriCommerce/rest-api/blob/master/acapi.yaml
- https://github.com/AmeriCommerce/rest-api/custom-properties

### PR DESCRIPTION
**Pull request by jcortes**
## WHY

Resolves #14198 

## Release Notes

- **New Features**
	- Introduced modules for managing order statuses, creating/updating orders, and updating customer details.
	- Added event handling for new customer and product creation, as well as new valid orders.
	- Enhanced API interaction with new constants and methods for retrieving customer and order data.
	- Implemented a webhook management module for creating and deleting webhooks.
	- Added a centralized source for event names related to payment processing, customer management, and order handling.

- **Bug Fixes**
	- Improved error handling for required fields during order creation.

- **Documentation**
	- Updated metadata and descriptions for new modules and methods.

- **Chores**
	- Updated package version and dependencies for better performance and compatibility.

### CODE REVIEW

File: components/americommerce/actions/change-order-status/change-order-status.mjs [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import app from "../../americommerce.app.mjs";
+
+export default {
+  key: "americommerce-change-order-status",
+  name: "Change Order Status",
+  description: "Changes the status of an existing order. Specify the order ID and the new status to update the order.",
```
[the remark this answers is not in the document]
Comment (GTFalcao) [reply components/americommerce/actions/change-order-status/change-order-status.mjs:6]: I don't think the second part of the description is needed. A docs link would be more useful instead

File: components/americommerce/actions/update-customer/update-customer.mjs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+      label: "Search String",
+      description: "The search string for the customer.",
+      optional: true,
+    },
+    noAccount: {
+      type: "boolean",
+      label: "No Account",
+      description: "Whether the customer has an account.",
+      optional: true,
+    },
```
[the remark this answers is not in the document]
Comment (GTFalcao) [reply components/americommerce/actions/update-customer/update-customer.mjs:94]: Agreed, I'd either change the label to "Has Account" or something of the like, or the description to reflect it, and adjust the boolean value accordingly.

File: components/americommerce/americommerce.app.mjs [hunk 3]
```diff
@@ -1,11 +1,253 @@
...
+          .map(({
+            first_name: firstName, last_name: lastName, id: value,
+          }) => ({
+            label: [
+              firstName,
+              lastName,
+            ].filter(Boolean).join(" "),
+            value,
+          }));
+      },
```
[the remark this answers is not in the document]
Comment (GTFalcao) [reply components/americommerce/americommerce.app.mjs:25]: Does this API support pagination? If so, agreed (not all do)

**Review by GTFalcao — COMMENTED** [review]
I left a few simple comments!

**Review by luancazarine — APPROVED** [review]
Hi @jcortes, I just added a comment but I'm moving it to Ready For QA anyway!

**Review by luancazarine — APPROVED** [review]

**Pull request merged**